### PR TITLE
Code insights: Implement delete insights with new dashboard functionality 

### PR DESCRIPTION
**Pull request by vovakulikov**
### Context

We always have this core insight functionality but with the dashboards system, we missed that at some point. This PR brings this functionality back. Currently, we have `delete` item in the insight context menu component. By click on that, you will delete this insight from all settings subjects and exclude this insight from all code insights dashboards. 

image

### Knowing issues
- Currently, if you delete some insight this operation triggers re-fetching all insights in the dashboard. This problem will be fixed in this ticket. https://github.com/sourcegraph/sourcegraph/issues/21097
- Delete operation doesn't have a confirmation UI (Related issues https://github.com/sourcegraph/sourcegraph/issues/21515) 

### What have been done 
- [x] Extract edit operations concept from the edit insight page 
- [x] Re-write use-delete-insight handler taking into account new code insight dashboard system
- [x] Connect handler and UI

### CODE REVIEW

File: client/web/src/insights/pages/insights/insights-page/hooks/delete-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            subjectId: configuredSubject.subject.id,
+            insightID: insightId,
+        }))
+
+    const removeInsightFromAllDashboards = settingsCascade.subjects.flatMap(configuredSubject => {
+        const { settings, subject } = configuredSubject
+
+        if (!settings || isErrorLike(settings)) {
+            return []
+        }
```
Comment (valerybugakov) [inline client/web/src/insights/pages/insights/insights-page/hooks/delete-helpers.ts:54]: nit: filter out them before both `removeInsightOperation` and `removeInsightFromAllDashboards` and reuse the same variable

File: client/web/src/insights/core/settings-action/edits/apply-edit-operations.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { SettingsOperation, SettingsOperationType } from './edits'
+
+/**
+ * Apply edit operation over jsonc settings string and return serialized final string
+ * with all applied edit operations.
+ *
+ * @param settings - original jsonc setting content
+ * @param operations - list of edit operations
+ */
+export function applyEditOperations(settings: string, operations: SettingsOperation[]): string {
```
Comment (valerybugakov) [inline client/web/src/insights/core/settings-action/edits/apply-edit-operations.ts:13]: Kudos for extracting helpers into smaller modules 🎉

**Review by valerybugakov — APPROVED** [review]
✅

File: client/web/src/insights/pages/insights/insights-page/hooks/delete-helpers.ts (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/insights/pages/insights/insights-page/hooks/delete-helpers.ts:54]: I tried to do that. The reason I left as it is - TS isn't smart enough to understand that we narrowed down nested properties 

Example: 

```ts
 const validSubjects = settingsCascade.subjects.filter(configuredSubject => {
    const { settings } = configuredSubject
    
    return settings && !isErrorLike(settings)
})

validSubjects.settings // still ErrorLike | null | Settings
```

I think some type guardы or type-casting may help here.

**Pull request merged**